pkg/operator/watchdog/proc.go

```diff
@@ -0,0 +1,57 @@
+package watchdog
+
+import (
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"syscall"
+)
+
+// FindPidByName find the process name specified by name and return the PID of that process.
+// If the process is not found, the bool is false.
+// NOTE: This require container with shared process namespace (if run as side-car).
+func FindPidByName(name string) (int, bool, error) {
+	files, err := ioutil.ReadDir("/proc")
+	if err != nil {
+		return 0, false, err
+	}
+	// sort means we start with the directories with numbers
+	sort.Slice(files, func(i, j int) bool {
+		return files[i].Name() < files[j].Name()
+	})
+	for _, file := range files {
+		if !file.IsDir() {
+			continue
+		}
+		// only scan process directories (eg. /proc/1234)
+		pid, err := strconv.Atoi(file.Name())
+		if err != nil {
+			continue
+		}
+		// read the /proc/123/exe symlink that points to a process
+		linkTarget := readlink(filepath.Join("/proc", file.Name(), "exe"))
+		if path.Base(linkTarget) != name {
+			continue
+		}
+		return pid, true, nil
+	}
+	return 0, false, nil
+}
+
+// readlink is copied from the os.Readlink() but does not return error when the target path does not exists.
+// This is used to read broken links as in case of share PID namespace, the /proc/1/exe points to a binary
+// that does not exists from the source container.
```

Comment:
omg

Reply:
https://github.com/golang/go/pull/31994 :-)

Reply:
Yeah, I've had very long discussion with @mfojtik about it, but there's no way, for now. He opened a go issue, iirc.